src/mokernels/lmm.jl

```diff
@@ -0,0 +1,46 @@
+@doc raw"""
+    NaiveLMMMOKernel(g, e::MOKernel, A::AbstractMatrix)
+
+Kernel associated with the linear mixing model.
+
+# Definition
+
+For inputs ``x, x'`` and output dimensions ``p_x, p_{x'}'``, the kernel is defined as[^BPTHST]
+```math
+k\big((x, p_x), (x, p_{x'})\big) = H_{p_{x}}K(x, x')H_{p_{x'}}
+```
+where ``K(x, x') = Diag(k_1(x, x'), ..., k_m(x, x'))`` with zero off-diagonal entries.
+``H_{p_{x}}`` is the ``p_x``-th column of ``H`` and
+``k_1, \ldots, k_m`` are ``m`` kernels, one for each latent process, ``H`` is a
+mixing matrix of ``m`` basis vectors spanning the output space.
+
+[^BPTHST]: Wessel P. Bruinsma, Eric Perim, Will Tebbutt, J. Scott Hosking, Arno Solin, Richard E. Turner (2020). [Scalable Exact Inference in Multi-Output Gaussian Processes](https://arxiv.org/pdf/1911.06287.pdf).
+"""
+struct NaiveLMMMOKernel{Tk<:AbstractVector{<:Kernel}, Th<:AbstractMatrix} <: MOKernel
+    K::Tk
+    H::Th
+    # if just a simple kernel is provided, we construct the MO kernel
+    function NaiveLMMMOKernel(k::SimpleKernel, H::AbstractMatrix)
+        m = size(H,2)
+        K = fill(k, m)
+        return new{typeof(K),typeof(H)}(K,H)
+    end
+    function NaiveLMMMOKernel(K::AbstractVector{<:Kernel}, H::AbstractMatrix)
+        return new{typeof(K),typeof(H)}(K,H)
+    end
+end
```

Comment:
I guess this could just be one outer constructor (the other one is just the default constructor)? And does not have to be restricted to `SimpleKernel`? And one could use `Fill` instead which does not allocate?

```suggestion
end

NaiveLMMMOKernel(k::Kernel, H::AbstractMatrix) = NaiveLMMMOKernel(Fill(k, size(H, 1)), H)
```

Reply:
Good idea, done!